packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+        komenci: {
+          ...feelessVerificationState.komenci,
+          sessionToken: komenciSessionResult.result,
+        },
+      })
+    )
+
+    // Fetch session state now that we are sure to have a token
+    yield call(fetchSessionState, komenciKit, e164Number)
+  }
+}
+
+// Checks if phoneHash is in cache, if it's not it asks Komenci to get it,
+// puts it in the cache, then checks again
+function* fetchPhoneHashDetails(komenciKit: KomenciKit, e164Number: string) {
+  Logger.debug(TAG, '@fetchPhoneHashDetails', 'Starting fetch')
+  try {
+    yield call(fetchPhoneHashDetailsFromCache, e164Number)
+  } catch (error) {
+    const feelessVerificationState: FeelessVerificationState = yield select(
+      feelessVerificationStateSelector
+    )
+
+    const blsBlindingClient = new ReactBlsBlindingClient(networkConfig.odisPubKey)
+    const pepperQueryResult: Result<GetDistributedBlindedPepperResp, FetchError> = yield call(
+      [komenciKit, komenciKit.getDistributedBlindedPepper],
+      e164Number,
+      DeviceInfo.getVersion(),
+      blsBlindingClient
+    )
+
+    if (!pepperQueryResult.ok) {
+      Logger.debug(TAG, '@fetchPhoneHashDetails', 'Unable to query for pepper')
+      throw pepperQueryResult.error
+    }
+
+    const pepperQuotaRemaining = feelessVerificationState.komenci.pepperQuotaRemaining - 1
+    yield put(
+      feelessUpdateVerificationState({
+        ...feelessVerificationState,
+        komenci: {
+          ...feelessVerificationState.komenci,
+          pepperQuotaRemaining,
+        },
+      })
+    )
+
+    yield put(updateE164PhoneNumberSalts({ [e164Number]: pepperQueryResult.result.pepper }))
+    yield call(fetchPhoneHashDetailsFromCache, e164Number)
+  }
+}
+
+function* fetchOrDeployMtw(
+  contractKit: ContractKit,
+  komenciKit: KomenciKit,
+  walletAddress: string,
+  e164Number: string
+) {
+  Logger.debug(TAG, '@fetchOrDeployMtw', 'Starting fetch')
+  let feelessVerificationState: FeelessVerificationState = yield select(
+    feelessVerificationStateSelector
+  )
+  let { unverifiedMtwAddress } = feelessVerificationState.komenci
+
+  // If there isn't a MTW for this session, ask Komenci to deploy one
+  if (!unverifiedMtwAddress) {
+    // This try/catch block is a workaround because Komenci will throw and error
+    // if a wallet was already deployed in a session. This is only fatal if
+    // we can't recover the MTW address or there is no quota left on the session
+    try {
+      const deployWalletResult: Result<string, FetchError | TxError | InvalidWallet> = yield call(
+        [komenciKit, komenciKit.deployWallet],
+        CURRENT_MTW_IMPLEMENTATION_ADDRESS
+      )
+
+      if (!deployWalletResult.ok) {
+        Logger.debug(TAG, '@fetchOrDeployMtw', 'Unable to deploy MTW')
+        throw deployWalletResult.error
+      }
+
+      unverifiedMtwAddress = deployWalletResult.result
+    } catch (error) {
+      // Fetch session state now that we are sure to have a MTW
```

Comment:
Why are we sure of this?

Reply:
Poorly worded comment on my part. Changed it to: `Fetch session state now that there should be no cases where we don't have a  MTW`.

As you can see below, I am checking if `unverifiedMtwAddress` is still undefined and throwing a fatal error if that's the case.